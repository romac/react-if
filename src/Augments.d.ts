declare namespace NodeJS {
  export interface ProcessEnv {
    NODE_ENV: 'development' | 'production' | 'test';
  }
}
