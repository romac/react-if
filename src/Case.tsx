import { FC } from 'react';
import { render } from './render';
import { ComponentWithConditionProps } from './types';

/** If the `<Case />` is the first one to have its condition evaluates to true inside the parent `<Switch />` it will be the only rendered. */
export const Case: FC<ComponentWithConditionProps> = props => render(props);

Case.defaultProps = {
  children: null
};
