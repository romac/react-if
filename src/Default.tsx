import { render } from './render';
import type { FCWithImplicitChildren } from './types';

/**
 * If no `<Case />` have its condition evaluates to true inside the parent `<Switch />`,
 * the first `<Default />` will be the only one rendered.
 * @param props The props to pass down to the `<Default />` component
 */
export const Default: FCWithImplicitChildren = (props) => render(props);

Default.defaultProps = {
  children: null
};
