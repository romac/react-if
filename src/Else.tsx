import { render } from './render';
import type { FCWithImplicitChildren } from './types';

/**
 * Must only contain a single child, which it renders as-is.
 * Should not be used outside of an `<If />` block.
 * @param props The props to pass down to the `<Else />` component
 */
export const Else: FCWithImplicitChildren = (props) => render(props);
