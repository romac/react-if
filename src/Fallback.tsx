import { render } from './render';
import type { FCWithImplicitChildren } from './types';

/**
 * Must contain only a single child, which it renders as-is.
 * Should not be used outside of an `<If />` block whose condition prop is a promise.
 * @param props The props to pass down to the `<Fallback />` component
 */
export const Fallback: FCWithImplicitChildren = (props) => render(props);
