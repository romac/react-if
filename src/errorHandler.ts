export const handleError = (errorMessage: string) => {
  if (process.env.NODE_ENV !== 'production') {
    throw new Error(errorMessage);
  }
};
