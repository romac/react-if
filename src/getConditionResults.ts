import { BooleanLike } from './types';

export const getConditionResult = (condition: BooleanLike | ((...args: unknown[]) => BooleanLike)) => {
  const conditionResult = Boolean(typeof condition === 'function' ? condition() : condition);

  return conditionResult;
};
