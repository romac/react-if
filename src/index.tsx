export * from './Case';
export * from './Default';
export * from './Else';
export * from './If';
export * from './Switch';
export * from './Then';
export * from './types';
export * from './Unless';
export * from './When';
