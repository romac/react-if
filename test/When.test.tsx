import { shallow } from 'enzyme';
import React from 'react';
import { When } from '../src';

describe('<When /> component', () => {
  describe('Truthy cases', () => {
    test('GIVEN some children THEN renders those', () => {
      const wrapped = shallow(
        <When condition={true}>
          <span>When</span>
        </When>
      );

      expect(wrapped).toMatchSnapshot();
      expect(wrapped.containsMatchingElement(<span>When</span>)).toBe(true);
    });

    test('GIVEN no children THEN renders null', () => {
      const wrapped = shallow(<When condition={true}></When>);

      expect(wrapped).toMatchSnapshot();
      expect(wrapped.html()).toBeNull();
    });

    test('GIVEN condition as function & children THEN renders those', () => {
      const wrapped = shallow(
        <When condition={() => true}>
          <span>When</span>
        </When>
      );

      expect(wrapped).toMatchSnapshot();
      expect(wrapped.containsMatchingElement(<span>When</span>)).toBe(true);
    });
  });

  describe('Falsy cases', () => {
    test('GIVEN some children THEN does not render those', () => {
      const wrapped = shallow(
        <When condition={false}>
          <span>When</span>
        </When>
      );

      expect(wrapped).toMatchSnapshot();
      expect(wrapped.containsMatchingElement(<span>When</span>)).toBe(false);
      expect(wrapped.html()).toBeNull();
    });

    test('GIVEN condition as function & some children THEN does not render those', () => {
      const wrapped = shallow(
        <When condition={() => false}>
          <span>When</span>
        </When>
      );

      expect(wrapped).toMatchSnapshot();
      expect(wrapped.containsMatchingElement(<span>When</span>)).toBe(false);
      expect(wrapped.html()).toBeNull();
    });
  });
});
