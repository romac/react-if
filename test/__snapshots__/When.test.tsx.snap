// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<When /> component Falsy cases GIVEN condition as function & some children THEN does not render those 1`] = `""`;

exports[`<When /> component Falsy cases GIVEN some children THEN does not render those 1`] = `""`;

exports[`<When /> component Truthy cases GIVEN condition as function & children THEN renders those 1`] = `
<Fragment>
  <span>
    When
  </span>
</Fragment>
`;

exports[`<When /> component Truthy cases GIVEN no children THEN renders null 1`] = `""`;

exports[`<When /> component Truthy cases GIVEN some children THEN renders those 1`] = `
<Fragment>
  <span>
    When
  </span>
</Fragment>
`;
